chain/chain/src/chain.rs

```diff
@@ -717,8 +717,12 @@ impl Chain {
                 {
                     return Err(ErrorKind::InvalidChunk.into());
                 }
-            } else {
-                if !runtime_adapter.verify_chunk_header_signature(&chunk_header.clone())? {
+            } else if chunk_header.height_created() == block.header().height() {
+                if !runtime_adapter.verify_chunk_header_signature(
+                    &chunk_header.clone(),
+                    block.header().epoch_id(),
+                    block.header().prev_hash(),
+                )? {
```

Comment:
Could we add a test for this?

Reply:
What kind of test do you mean? I can also just keep the code behavior here the same if you want.

Reply:
> This PR also fixes another bug in store_partial_encoded_chunk about verifying chunk header signature. It was wrong before because it uses block producer instead of chunk producer.

I am not sure what the bug is actually. Could you add a test for that?

Reply:
Ah i c, the bug was not related to this line. I just refactored the code on `verify_chunk_header_signature`, which was related to the bug.